tests/simulate_data/test_get_simulated_data.py

```diff
@@ -0,0 +1,354 @@
+"""
+test_simulated_data_new::simulate_data
+
+We test :
+- The dimensions of the outputs
+- Whether they should be binary or not
+- Whether the effects are coherent
+- Whether forbidden inputs return an error
+
+We pinpoint aberrant behavior reguarding some input combinaisons
+
+Reminder :
+p_t = P(T=1|X)
+th_p_t_mx = P(T=1|X,M)
+"""
+
+from pprint import pprint
+import pytest
+import numpy as np
+from numpy.random import default_rng
+from med_bench.src.get_simulated_data import simulate_data
+
+
+@pytest.fixture(autouse=True)
+def data():
+    n = 3
+    rg = default_rng(42)
+    mis_spec_m = False
+    mis_spec_y = False
+    dim_x = 1
+    dim_m = 1
+    seed = 1
+    type_m = "binary"
+    sigma_y = 0.5
+    sigma_m = 0.5
+    beta_t_factor = 10
+    beta_m_factor = 1
+
+    return simulate_data(
+        n,
+        rg,
+        mis_spec_m,
+        mis_spec_y,
+        dim_x,
+        dim_m,
+        seed,
+        type_m,
+        sigma_y,
+        sigma_m,
+        beta_t_factor,
+        beta_m_factor,
+    )
+
+
+@pytest.fixture(autouse=True)
+def effects():
+    return np.array(data[4:9])
+
+
+def test_dimension_x(n, dim_x):
+    assert (data[0]).shape == (n, dim_x)
+
+
+def test_dimension_t(n):
+    assert (data[1]).shape == (n, 1)
+
+
+def test_dimension_m(n, dim_m):
+    assert (data[2]).shape == (n, dim_m)
+
+
+def test_dimension_y(n):
+    assert (data[3]).shape == (n, 1)
+
+
+def test_m_is_binary(type_m):
+    if type_m == "binary":
+        for m in data[2]:
+            assert m in {0, 1}
+    else:
+        for m in data[2]:
+            assert not m in {0, 1}
+
+
+def test_total_is_direct_plus_indirect():
+    assert effects[0] == effects[1] + effects[4]  # total = theta_1 + delta_0
+    assert effects[0] == effects[2] + effects[3]  # total = theta_0 + delta_1
+
+
+def test_effects_are_equals_if_y_well_specified(mis_spec_y):
+    if mis_spec_y:
+        assert effects[1] != effects[2]
+        assert effects[3] != effects[4]
+    else:
+        assert effects[1] == effects[2]
+        assert effects[3] == effects[4]
+
+
+# n=0 : Warnings
+@pytest.mark.xfail
+def test_n_null_should_fail():
+    with pytest.raises(Exception):
+        simulate_data(
+            n=0,
+            rg=default_rng(42),
+            mis_spec_m=False,
+            mis_spec_y=False,
+            dim_x=1,
+            dim_m=1,
+            seed=1,
+            type_m="binary",
+            sigma_y=0.5,
+            sigma_m=0.5,
+            beta_t_factor=1,
+            beta_m_factor=1,
+        )
+
+
+# n<0 : l19 ; ValueError: negative dimensions are not allowed
+@pytest.mark.xfail
```

Comment:
it's probaby worth trying parametrized tests, as those tests are all in the same line (validity of some parameter for sizes). The full list of parameters can be long, but something in this line should do the trick (more pseudo code for the idea than actual working code)
```python
@pytest.mark.parametrize("failing_value", [
    {n: 0},
    {n: -1},
], ids=[
    "n_null",
    "negative_n",
])
def test_timedistance_v0(value):
    default_values= {
        n: 2,
        dim_x: 5,
    }
    default_values.update(value)
    with pytest.raise(ValueError):
        simulate_data(**default_values)
```

Reply:
I will implement parametrization for this test in a future PR